src/Services/ShopService/Shopify.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+
+namespace Tychovbh\Mvc\Services\ShopService;
+
+
+use GuzzleHttp\Client;
+use Illuminate\Support\Arr;
+
+class Shopify implements ShopServiceInterface
+{
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Retrieves all products
+     * @param array $params
+     * @return array
+     */
+    public function products(array $params = []): array
+    {
+        $response = $this->request('get', 'products', '2020-10', $params);
+
+        return array_map(function ($product) {
+            return [
+                'id' => Arr::get($product, 'id'),
+                'title' => Arr::get($product, 'title'),
+                'price' => Arr::get($product, 'variants.0.price')
+            ];
+        }, Arr::get($response, 'products'));
+    }
+
+    /**
+     * Retrieves product by id
+     * @param int $id
+     * @return array
+     */
+    public function productById($id): array
+    {
+        $response = $this->request('get', 'products/' . $id, '2020-10');
+
+        return [
+            'id' => Arr::get($response, 'product.id'),
+            'title' => Arr::get($response, 'product.title'),
+            'price' => Arr::get($response, 'product.variants.0.price'),
+        ];
+    }
+
+    /**
+     * Retrieves all orders
+     * @param array $params
+     * @return array
+     */
+    public function orders(array $params = []): array
+    {
+        $response = $this->request('get', 'orders', '2020-10', $params);
+
+        return array_map(function ($order) {
+            return [
+                'id' => Arr::get($order, 'id'),
+                'email' => Arr::get($order, 'email'),
+                'discount_codes' => array_map(function ($discount_code) {
+                    return Arr::get($discount_code, 'code');
+                }, Arr::get($order, 'discount_codes')),
```

Comment:
add default empty array here in Arr::get, if discount_codes does not exists this code will break